cms/src/components/content/ContentForm.vue

```diff
@@ -171,9 +170,25 @@ const hasTag = computed(() => {
 });
 
 const isDirty = ref(false);
+const isEditingSlug = ref(false);
+const slugInput = ref<HTMLInputElement | undefined>(undefined);
+
+const updateSlug = async (text: string) => {
+    if (!text.trim() && values.title) text = values.title;
+    if (!text.trim()) return;
+    setValues({ slug: await Slug.generate(text.trim(), props.content._id) });
+};
 
-const updateSlug = async (title: string) => {
-    setValues({ slug: await Slug.generate(title, props.content._id) });
+let previousTitle: string = values.title || "";
+const updateSlugFromTitle = async (title: string) => {
+    // Check if the slug is still the default value
+    if (
+        previousTitle == "" ||
+        values.slug?.replace(/-[0-9]*$/g, "") == Slug.generateNonUnique(previousTitle)
+    ) {
+        await updateSlug(title);
+    }
+    previousTitle = title;
 };
```

Comment:
Not sure to what degree we need to test the slug functionality?

Reply:
Would be nice to have a test that it doesn't update the slug if you update the title and the slug has been modified.

Reply:
Busy writing it now